unit-tests/py/rspy/devices.py

```diff
@@ -599,11 +599,14 @@ def hw_reset( serial_numbers, timeout = MAX_ENUMERATION_TIME ):
     :param timeout: Maximum # of seconds to wait for the devices to come back online
     :return: True if all devices have come back online before timeout
     """
+
+    usb_serial_numbers = {dev.serial_number for dev in _device_by_sn.values() if dev.port is not None and dev.serial_number in serial_numbers}
```

Comment:
This will work.
But seems kind of unneeded since _device_by_sn is already keyed to SNs:
`usb_serial_numbers = {sn for sn in serial_numbers if _device_by_sn[sn].port is not None}`
This will also throw if given a SN that's invalid. Good IMO.

Reply:
Done